spec/services/hyrax/embargo_manager_spec.rb

```diff
@@ -220,6 +220,35 @@
     end
   end
 
+  describe '#release_embargo_for' do
+    let(:manager) { described_class }
+
+    context 'with an expired embargo' do
+      include_context 'with expired embargo'
+      let!(:vis_after_embargo) { embargo.visibility_after_embargo }
+
+      it 'removes the embargo from the resource' do
+        manager.release_embargo_for(resource: resource)
+
+        # #release will set the visibility correctly on the resource
+        # #nullify however will then set embargo.visibility_after_embargo to be nil
+        expect(resource.visibility).to eq(vis_after_embargo)
+        expect(resource.embargo.embargo_release_date).to be nil
+        expect(resource.embargo.visibility_during_embargo).to be nil
```

Comment:
is this the initial problem?

i recall thinking at initial implementation time that it was simpler and probably desirable in general to leave this data in place when releasing embargoes. since the embargo is a separate Resource, it seemed just destructive to wipe its data.

Reply:
I'm unsure whether this is actually the initial problem, but I know it solves the problem. 

without calling `#nullify`, the resource's visibility would say it was updated, but the embargo wasn't actually removed. refreshing the ui would still show the work, just in a different order. (e.g. it may have initially been the first work on "/embargoes", but now it's the last). I added the nullify because it follows the pattern that `deactivate!` does.